src/components/DiskSource/DiskSource.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { V1alpha1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  Checkbox,
+  FormGroup,
+  Popover,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextInput,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import { PersistentVolumeClaimSelect } from '../PersistentVolumeClaimSelect';
+
+import {
+  DISK_SOURCE_OPTIONS,
+  HTTP_DISK_SOURCE_NAME,
+  PVC_DISK_SOURCE_NAME,
+  PVC_SIZE_FORMATS,
+  REGISTRY_DISK_SOURCE_NAME,
+} from './constants';
+import { getPVCDiskSource } from './utils';
+import { VolumeSize } from './VolumeSize';
+import { DISK_SOURCE_OPTIONS_IDS } from '.';
+
+import './DiskSource.scss';
+
+type DiskSourceProps = {
+  template: V1Template;
+  onChange: (diskSource: any) => void;
+};
+
+export const DiskSource: React.FC<DiskSourceProps> = ({ onChange }) => {
+  const { t } = useKubevirtTranslation();
+
+  const [urlText, setURLText] = React.useState('');
+  const [pvcSize, setPVCSize] = React.useState(30);
+  const [pvcSizeFormat, setPVCSizeFormat] = React.useState(PVC_SIZE_FORMATS.GiB);
+
+  const [selectedSourceType, setSourceType] =
+    React.useState<DISK_SOURCE_OPTIONS_IDS>(PVC_DISK_SOURCE_NAME);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const onSelectDiskSource = React.useCallback((event, selection) => {
+    setSourceType(DISK_SOURCE_OPTIONS.find((source) => source.name === selection)?.name);
+    setIsOpen(false);
+  }, []);
+
+  const diskSourceLabel = (
+    <>
+      {t('Disk source')}{' '}
+      <Popover
+        aria-label={t('Help')}
+        bodyContent={() => (
+          <div>
+            {t(
+              'Disk Source represents the source for our Disk, this can be HTTP, Registry or an excisting PVC',
+            )}
+          </div>
+        )}
+      >
+        <HelpIcon />
+      </Popover>
+    </>
+  );
+
+  const onPVCChange = React.useCallback(
+    (pvc: V1alpha1PersistentVolumeClaim) => {
+      onChange(getPVCDiskSource(pvc));
+    },
+    [onChange],
+  );
+
+  return (
+    <>
+      <FormGroup
+        label={diskSourceLabel}
+        fieldId={`disk-source-required-disk`}
+        isRequired
+        className="disk-source-form-group"
+      >
+        <Select
+          isOpen={isOpen}
+          onToggle={setIsOpen}
+          onSelect={onSelectDiskSource}
+          variant={SelectVariant.single}
+          selections={selectedSourceType}
+        >
+          {DISK_SOURCE_OPTIONS.map((option) => (
```

Comment:
IMHO it's more readable when it's inline,
how may times `DISK_SOURCE_OPTIONS` is used ? can we do without it ?

Reply:
okay thanks :D